### PR TITLE
Add com.synology.SynologyDrive

### PR DESCRIPTION
**Pull request by gregos-winus**
Desktop application to synchronize files and folders between the computer and the Synology Drive server to access, browse, and share files via file browse
Real-time or scheduled computer backup from the local computer to Synology Drive Server with the capability to restore files and folders on the client side
https://www.synology.com/en-us/dsm/feature/drive

### CODE REVIEW

File: com.synology.SynologyDrive.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com.synology.SynologyDrive</id>
+  <metadata_license>MIT</metadata_license>
+  <project_license>LicenseRef-proprietary=https://code.visualstudio.com/license</project_license>
```
Comment (AsciiWolf) [inline com.synology.SynologyDrive.metainfo.xml:5]: Is the license url really the correct one? :-)

Comment (AsciiWolf) [inline com.synology.SynologyDrive.metainfo.xml:5]: Is the license url really the correct one? :-)

Comment (gregos-winus) [reply com.synology.SynologyDrive.metainfo.xml]: sorry... it's not the correct license url. i used visual studio code as a template and i missed that line. sorry.
it's now corrected in the new commits

Comment (AsciiWolf) [reply com.synology.SynologyDrive.metainfo.xml]: No problem, that's what the new app review process is for. :-) Anyway, the license url is still not correct.

Comment (gregos-winus) [reply com.synology.SynologyDrive.metainfo.xml]: i've the good url in the new commit : 43e61b0a7e50cabddac1cce3d2dd037e52bf61da

Comment (gregos-winus) [reply com.synology.SynologyDrive.metainfo.xml]: i think it's OK no, in the commit effbcde
(the good url is [https://www.synology.com/en-global/company/legal/terms_EULA](https://www.synology.com/en-global/company/legal/terms_EULA))

Comment (AsciiWolf) [reply com.synology.SynologyDrive.metainfo.xml]: Yep, it's fine now.

**Comment by LeandroStanger** [comment]
bot, build com.synology.SynologyDrive

File: com.synology.SynologyDrive.desktop [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Synology Drive Client
+Comment=Synology Drive Client
+Exec=synology-drive start
+Icon=me.gregos.SynologyDrive
```
Comment (nedrichards) [inline com.synology.SynologyDrive.desktop:5]: this is not the icon you're installing. it should be `com.synology.SynologyDrive`

File: com.synology.SynologyDrive.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com.synology.SynologyDrive</id>
+  <project_license>LicenseRef-proprietary=https://www.synology.com/en-global/company/legal/terms_EULA</project_license>
+  <name>Synology Drive</name>
+  <developer_name>Synology</developer_name>
+  <url type="homepage">https://www.synology.com</url>
+  <summary>Powerful private cloud storage with no recurring fees</summary>
+  <description>
+    <p>Now, Synology makes it possible for every home and office to host your own cloud – with 100% data ownership and no subscription fees. See how this changes the way we live and work, access and share data.</p>
```
Comment (nedrichards) [inline com.synology.SynologyDrive.metainfo.xml:10]: are you from Synology? If not can you add a disclaimer like `NOTE: This wrapper is not verified by, affiliated with, or supported by Synology Inc.` as a new paragraph here.

**Comment by nedrichards** [comment]
Can you see if this can be integrated with the external data checker as mentioned here? https://github.com/flathub/flathub/wiki/App-Requirements#external-data-checker

**Comment by nedrichards** [comment]
This needs to add a flathub.json file to prevent arm and aarch64 from being distributed. An example is here https://github.com/flathub/org.kde.skrooge/blob/master/flathub.json (you only want x86_64 of course)

File: com.synology.SynologyDrive.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        path: icons/synologydrive-256.png
+      - type: extra-data
+        filename: synology-drive.deb
+        only-arches: [x86_64]
+        url: https://global.download.synology.com/download/Tools/SynologyDriveClient/2.0.1-11061/Ubuntu/Installer/x86_64/synology-drive-client-11061.x86_64.deb
+        sha256: c9a55d9debc6e9521b4f465a5dd55aeb7d2eec7d091658a8f40cb360d71d5edf
+        size: 111162324
+      - type: extra-data
+        filename: synology-drive.deb
+        only-arches: [i686]
```
Comment (nedrichards) [inline com.synology.SynologyDrive.yaml:55]: the 19.08 runtime no longer supports 32bit x86 applications, you should remove this section.

File: com.synology.SynologyDrive.desktop (hunk 2 above)
Comment (gregos-winus) [reply com.synology.SynologyDrive.desktop:5]: corrected for the next commit

File: com.synology.SynologyDrive.metainfo.xml (hunk 3 above)
Comment (gregos-winus) [reply com.synology.SynologyDrive.metainfo.xml:10]: ok

File: com.synology.SynologyDrive.yaml (hunk 4 above)
Comment (gregos-winus) [reply com.synology.SynologyDrive.yaml:55]: ok

**Comment by gregos-winus** [comment]
All nedrichards' comments are implemented except the external data checker. 

> Can you see if this can be integrated with the external data checker as mentioned here? https://github.com/flathub/flathub/wiki/App-Requirements#external-data-checker

IMO, **it is not possible** because :

- on synology's website, there is no generic link to download the latest version of the software ( in their download center, in all pages, download links contains the version number)
=> so **we can not implement the rotating-url process** of the external data checker

- the software version is defined by a version number and a release number. in download link, you can find those number in 2 different formats: "...../$versionnumber-$releasenumber..../" and ".......$releasenumber.x86_64.deb"
in the example below, versionnumber=2.0.1 and releasenumber=11078
https://global.download.synology.com/download/Tools/SynologyDriveClient/2.0.2-11078/Ubuntu/Installer/x86_64/synology-drive-client-11078.x86_64.deb
**as far as the html checker can use just only one var and regex, it can not be implemented for that software**

- there is **no debian repo** hosted by synology

**Comment by nedrichards** [comment]
bot, build com.synology.SynologyDrive

**Comment by gregos-winus** [comment]
> Build [19932 failed](https://flathub.org/builds/#/builders/32/builds/19932)

Error fixed (missing metainfo.xml in the manifest

**Comment by bilelmoussaoui** [comment]
bot, build com.synology.SynologyDrive

**Comment by nedrichards** [comment]
@gregos-winus OK, i've run this test build locally with dbus logging and there's some more names you need to whitelist. You can test this yourself with `flatpak run --log-session-bus com.synology.SynologyDrive` based on this I think you need to add:

> --talk-name=org.mpris.MediaPlayer2.com.synology.SynologyDrive
> --talk-name=org.freedesktop.Notifications
> --talk-name=org.a11y.*

**Comment by nedrichards** [comment]
note: I certainly made a mistake there, it should be --own-name for the mpris socket. Anyway, double check everything with the dbus logger.

**Comment by bilelmoussaoui** [comment]
I don't think it needs the MediaPlayer nor org.a11y.* access, they are given by default, running the application with --log-session-bus shows at first the dbus access that you have before complaining about missing ones. It only needs the notifications one it seems

**Comment by nedrichards** [comment]
thanks @bilelmoussaoui I've been using that log for ever and reading it still trips me up ;-) I've filed https://github.com/flatpak/flatpak/issues/3613 to discuss better tooling.

**Comment by gregos-winus** [comment]
> @gregos-winus OK, i've run this test build locally with dbus logging and there's some more names you need to whitelist. You can test this yourself with `flatpak run --log-session-bus com.synology.SynologyDrive` based on this I think you need to add:
> 
> > --talk-name=org.mpris.MediaPlayer2.com.synology.SynologyDrive
> > --talk-name=org.freedesktop.Notifications
> > --talk-name=org.a11y.*



> I don't think it needs the MediaPlayer nor org.a11y.* access, they are given by default, running the application with --log-session-bus shows at first the dbus access that you have before complaining about missing ones. It only needs the notifications one it seems

added "--talk-name=org.freedesktop.Notifications" in manifest, the software behaves the same way than before (without it).

File: com.synology.SynologyDrive.yaml [hunk 5]
```diff
@@ -51,6 +52,7 @@ modules:
       - type: extra-data
         filename: synology-drive.deb
         only-arches: [x86_64]
-        url: https://global.download.synology.com/download/Tools/SynologyDriveClient/2.0.2-11078/Ubuntu/Installer/x86_64/synology-drive-client-11078.x86_64.deb
+        url: http://192.168.2.2/synology-drive.deb
```
Comment (bilelmoussaoui) [inline com.synology.SynologyDrive.yaml]: that doesn't look like the correct url

Comment (gregos-winus) [reply com.synology.SynologyDrive.yaml]: i know and that's why i already pushed a commit (that correct the typo)
i've a very slow internet connection and that's my workaround when i run "flatpak-builder _blablablabla_"

**Comment by Dricc123** [comment]
@gregos-winus indeed, I see there is no autostart.
I also noticed following bug. Do you confirm it?
steps:
I exit the application.
I re-enter the application: it re-initiates, the task that I created to synch some folders has disappeared (I need to re-create the task to resync the files again)

**Comment by Dricc123** [comment]
I'm starting to think it is a SynologyDrive bug rather than a Flatpak one. from what I see here, other people experience sync tasks being deleted https://community.synology.com/enu/forum/1/post/138397

File: com.synology.SynologyDrive.yaml (hunk 4 above)
Comment (Bekon241) [reply com.synology.SynologyDrive.yaml]: For flatpak version just add to demon. (startup programs)

name: Synology Drive Client

command: /usr/bin/flatpak run --branch=stable --arch=x86_64 --command=synology-drive com.synology.SynologyDrive start

Info: Synchronize files in cloud.

Tested on Fedora works. Cheers from Poland.

**Pull request closed without merging**